citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java

```diff
@@ -2472,6 +2486,34 @@ public String buildNewResourcePanelContext(VelocityPortlet portlet, Context cont
 			contentProperties = new HashMap<String,Object>();
 		}
 		context.put("contentProperties", contentProperties);
+
+		try
+		{
+			ContentResource cr = contentService.getResource(MANAGING_LIBRARIES_JSON);
+			BufferedReader streamReader = new BufferedReader(new InputStreamReader(cr.streamContent(), "UTF-8"));
+			StringBuilder responseStrBuilder = new StringBuilder();
+
+			String inputStr;
+			while ((inputStr = streamReader.readLine()) != null){
+				responseStrBuilder.append(inputStr);
+			}
+			JSONObject jsonObject = JSONObject.fromObject(responseStrBuilder.toString());
+			JSONArray managingLibraries = jsonObject.getJSONArray("managingLibraries");
+			context.put("managingLibraries", managingLibraries);
+		}
+		catch (IdUnusedException e)
+		{
+			logger.debug("The managingLibraries.json file doesn't exist", e);
+		}
+		catch (TypeException e)
+		{
+			logger.debug("The managingLibraries.json file doesn't exist", e);
+		}
+		catch (Exception e)
```

Comment:
Do you really need to catch `Exception` here? In general it makes the code harder to read as it's not clear what things you expect to go wrong (eg NullPointerExceptions, ArrayOutOfBoundsExceptions).